libs/vkd3d/resource.c

```diff
@@ -3344,22 +3348,26 @@ void d3d12_rtv_desc_create_rtv(struct d3d12_rtv_desc *rtv_desc, struct d3d12_dev
                 vkd3d_desc.miplevel_idx = desc->u.Texture2DArray.MipSlice;
                 vkd3d_desc.layer_idx = desc->u.Texture2DArray.FirstArraySlice;
                 vkd3d_desc.layer_count = desc->u.Texture2DArray.ArraySize;
+                vkd3d_desc.layer_count = min(vkd3d_desc.layer_count, resource->desc.DepthOrArraySize - vkd3d_desc.layer_idx);
```

Comment:
I think this would be more readable if there this came after the switch block since it applies to all arrayed types and shouldn't have any effect on non-arrayed types, or as `min(desc->u.Texture2DArray.ArraySize, resource->desc.DepthOrArraySize - vkd3d_desc.layer_idx)` etc.

Reply:
Fixed.